homeassistant/components/light/__init__.py

```diff
@@ -228,7 +231,12 @@ def preprocess_turn_on_alternatives(params):
 
     color_name = params.pop(ATTR_COLOR_NAME, None)
     if color_name is not None:
-        params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
+        try:
+            params[ATTR_RGB_COLOR] = color_util.color_name_to_rgb(color_name)
+        except ValueError:
```

Comment:
`color_name_to_rgb` will now raise instead of silently returning white if no match found. I've added that logic back in the only place that depended on it.